api/v1/nodescan/service.go

```diff
@@ -238,7 +240,31 @@ func (s *serviceImpl) getRuntimeVulns(containerRuntime *v1.GetNodeVulnerabilitie
 	return nil, nil
 }
 
-func (s *serviceImpl) GetNodeVulnerabilities(_ context.Context, req *v1.GetNodeVulnerabilitiesRequest) (*v1.GetNodeVulnerabilitiesResponse, error) {
+func (s *serviceImpl) GetNodeVulnerabilities(ctx context.Context, req *v1.GetNodeVulnerabilitiesRequest) (*v1.GetNodeVulnerabilitiesResponse, error) {
+	// If NodeInventory is empty `req.GetComponents() == nil` then fallback to v1 scanning
+	if req.GetComponents() == nil || !featureFlags.RHCOSNodeScanning.Enabled() {
+		return s.getNodeVulnerabilitiesLegacy(ctx, req)
+	}
+
+	resp := &v1.GetNodeVulnerabilitiesResponse{
+		ScannerVersion: s.version,
+	}
+
+	if !wellknownnamespaces.IsRHCOSNamespace(req.GetComponents().GetNamespace()) || len(req.GetComponents().GetRhelContentSets()) == 0 {
+		// Non-RHCOS system with node inventory detected!
+		// We can provide a list of pkgs but cannot scan them -> setting a note to inform the user.
```

Comment:
Should we update this comment?

```suggestion
		// Node components exist in the request, but we cannot perform vulnerability matching
		// if (a.) its not an RHCOS namespaces or, (b.) if it is an RHCO namespace missing content
		// sets. The latter occurs for old RHCO versions we don't support.
```

Reply:
I think ^ is missing an S in RHCOS :)